diff --git a/src/rutabaga_gralloc/minigbm.rs b/src/rutabaga_gralloc/minigbm.rs
index 0d8127a..9676bf2 100644
--- a/src/rutabaga_gralloc/minigbm.rs
+++ b/src/rutabaga_gralloc/minigbm.rs
@@ -12,7 +12,6 @@ use std::fs::File;
 use std::io::Error;
 use std::io::Seek;
 use std::io::SeekFrom;
-use std::os::fd::FromRawFd;
 use std::sync::Arc;
 
 use magma_gpu::util::Error as MagmaGpuError;
@@ -25,6 +24,8 @@ use crate::rutabaga_gralloc::gralloc::Gralloc;
 use crate::rutabaga_gralloc::gralloc::ImageAllocationInfo;
 use crate::rutabaga_gralloc::gralloc::ImageMemoryRequirements;
 use crate::rutabaga_gralloc::minigbm_bindings::*;
+use crate::rutabaga_gralloc::rendernode;
+use magma_gpu::util::AsRawDescriptor;
 use crate::rutabaga_utils::RutabagaError;
 use crate::rutabaga_utils::RutabagaResult;
 use crate::rutabaga_utils::RUTABAGA_MAP_CACHE_CACHED;
@@ -57,26 +58,22 @@ impl Drop for MinigbmDeviceInner {
 pub struct MinigbmDevice {
     minigbm_device: Arc<MinigbmDeviceInner>,
     last_buffer: Option<Arc<MinigbmBuffer>>,
+    device_name: String,
 }
 
 impl MinigbmDevice {
     /// Returns a new `MinigbmDevice` if there is a rendernode in `/dev/dri/` that is accepted by
     /// the minigbm library.
     pub fn init() -> RutabagaResult<Box<dyn Gralloc>> {
-        let descriptor: File;
-        let gbm: *mut gbm_device;
-        // SAFETY:
-        // Safe because minigbm_create_default_device is safe to call with an unused fd,
-        // and fd is guaranteed to be overwritten with a valid descriptor when a non-null
-        // pointer is returned.
-        unsafe {
-            let mut fd = -1;
+        let undesired: &[&str] = &["vgem", "pvr"];
+        let (descriptor, device_name) = rendernode::open_device(undesired)?;
 
-            gbm = minigbm_create_default_device(&mut fd);
-            if gbm.is_null() {
-                return Err(MagmaGpuError::IoError(Error::last_os_error()).into());
-            }
-            descriptor = File::from_raw_fd(fd);
+        // SAFETY:
+        // gbm_create_device is safe to call with a valid fd, and we check that a valid one is
+        // returned.  If the fd does not refer to a DRM device, gbm_create_device will reject it.
+        let gbm = unsafe { gbm_create_device(descriptor.as_raw_descriptor()) };
+        if gbm.is_null() {
+            return Err(MagmaGpuError::IoError(Error::last_os_error()).into());
         }
 
         Ok(Box::new(MinigbmDevice {
@@ -85,6 +82,7 @@ impl MinigbmDevice {
                 gbm,
             }),
             last_buffer: None,
+            device_name,
         }))
     }
 }
@@ -123,7 +121,7 @@ impl Gralloc for MinigbmDevice {
             _device: self.clone(),
         };
 
-        if gbm_buffer.cached() {
+        if self.device_name == "i915" || self.device_name == "xe" || info.flags.host_cached() {
             reqs.map_info = RUTABAGA_MAP_CACHE_CACHED;
         } else {
             reqs.map_info = RUTABAGA_MAP_CACHE_WC;
@@ -259,13 +257,6 @@ impl MinigbmBuffer {
         unsafe { gbm_bo_get_stride_for_plane(self.bo, plane) }
     }
 
-    /// Should buffer use cached mapping to guest
-    pub fn cached(&self) -> bool {
-        // SAFETY:
-        // This is always safe to call with a valid gbm_bo pointer.
-        let mode = unsafe { gbm_bo_get_map_info(self.bo) };
-        mode == gbm_bo_map_cache_mode::GBM_BO_MAP_CACHE_CACHED
-    }
 
     /// Exports a new dmabuf/prime file descriptor.
     pub fn export(&self) -> RutabagaResult<File> {
